compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -617,8 +617,7 @@ impl<'a> AstValidator<'a> {
             .session
             .source_map()
             .span_to_snippet(span)
-            .map(|snippet| snippet.starts_with("#["))
-            .unwrap_or(true);
+            .map_or(true, |snippet| snippet.starts_with("#["));
         if !is_macro_callsite {
```

Comment:
Since this condition is only used inverted, it might be easier to read to just use `is_some_and` with the inverted condition:

```rust
        if self
            .session
            .source_map()
            .span_to_snippet(span)
            .is_some_and(|snippet| !snippet.starts_with("#["))
        {
```

Then it clearly reads as "if the snippet is available and doesn't start with `#[`".

Reply:
Good catch, thanks!

(as a side note `span_to_snippet` returns a result, so this would have to be a `is_ok_and`, but yeah)